### ISSUE TITLE
when to call grad.data.zero_()?

### ISSUE THREAD

**Issue by abhigenie92** [llSourcell/pytorch_in_5_minutes#1]
1. Why did you call  w1.grad.data.zero_() after loss.backward() and not before for the first iteration?
2. Is it required?
